base/dates/io.jl

```diff
@@ -152,7 +152,7 @@ function parse(x::AbstractString,df::DateFormat)
         throw(ArgumentError("Delimiter mismatch. Couldn't find first delimiter, \"$(df.slots[1].transition)\", in date string"))
     end
     periods = Period[]
-    extra = []
+    extra = Period[]
```

Comment:
If @omus agree, I would argue that extra is unused. If used, I guess one would have to declare new date constructors. I would propose to remove it, as in https://github.com/felipenoris/julia/commit/291cf0f568039369d5ad7184730e96ab88299efe . Possibly a breaking change.